Assets/Runtime/core/GameManager.cs

```diff
@@ -1,18 +1,11 @@
 ﻿using System;
 using UnityEngine;
 
-public class GameManager : MonoBehaviour {
-  private static GameManager manager = null;
+public class GameManager : SingletonMonoBehavior<GameManager> {
   private GameState state = GameState.ACTIVE;
 
   private void Awake() {
```

Comment:
This Awake method is overriding the inherited `Awake()` from `SingletonMonoBehavior`. That means your instance check and deleting if duplicate will never run. It is then possible for multiple instances of GameManger to exist and be active in the scene. Only the first will be registered as the singleton instance, but the rest will still be active and ticking on Update

One solution would be to set `SingletonMonoBehavior`'s `Awake` method to `virtual`, and then have `GameManager`'s `override` and call the base Awake.

Reply:
I missed that Unity private methods inherit, unlike normal C# private methods. Will fix per your suggestion, then make sure Unity won't e.g. ignore the override and call both Awake methods.

Reply:
Fixed.